tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -200,13 +200,29 @@ public async Task HandleRecordRequestAsync(string recordingId, HttpRequest incom
 
             HttpResponseMessage upstreamResponse = null;
 
+            // We don't want to erroneously send a Content-Length header if one is not present. The reason this is happening is that the outgoing response we are modifying
+            // is already initialized as if a body were going to be present. This is due to the fact that ASP.NET doesn't KNOW if there going to be response yet, and having everything
+            // already initialized and ready to be customized is definitely the efficient way to go.
+            // 
+            // To properly account for that, we need to _null out_ the Content property to actually allow the .NET httpclient to send without the added header.
+            if (!incomingRequest.Headers.ContainsKey("Content-Length"))
+            {
+                upstreamRequest.Content = null;
```

Comment:
I actually tried _removing_ the content-length header via `upstreamRequest.Content.Headers.Remove('Content-Length')` 

However, when you do it this way, ASP.NET still thinks there is a _body_ (because upstreamRequest.Content is not null), so it _re-adds_ the content-length header. There _is_ a body, it's just 0 length!

By setting the actual `Content` to null, the httpclient realizes that there ISNT any content, and it DOESNT introduce a `Content-Length: 0` as a result.

Reply:
Ok. So `Content-Length` is _not_ required when `Transfer-Encoding` is set to `chunked`. However, the `Content-Type` should _still_ be set. Our check here checks for _both_ being null. I think this should be safe based on that. 

[Mozilla Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding)